File: Genérica2.java
class Tres<T> {
    private T primero;
    private T segundo;
    private T tercero;

    public Tres(T primero, T segundo, T tercero) {
        this.primero = primero;
        this.segundo = segundo;
        this.tercero = tercero;
    }

    public T getPrimero() {
        return primero;
    }

    public T getSegundo() {
        return segundo;
    }

    public T getTercero() {
        return tercero;
    }
}

public class Genérica2 {
    public static void main(String[] args) {
        Tres<Integer> parEnteros = new Tres<>(1, 2, 3);
        System.out.println("Primero: " + parEnteros.getPrimero() + ", Segundo: " + parEnteros.getSegundo() + "Tercero: " + parEnteros.getTercero());

        Tres<String> parCadenas = new Tres<>("Para entender ", "la historia de ", "FNAF");
        System.out.println("Primero: " + parCadenas.getPrimero() + ", Segundo: " + parCadenas.getSegundo() + "Tercero: " + parCadenas.getTercero());
    }
}

File: Genérica2.cpp
#include <iostream>
#include <string>
#include <vector>

template<typename T>
T retornarValor(const T& valor) {
    return valor;
}

int main() {
    std::cout << retornarValor(123) << std::endl;

    std::cout << retornarValor(std::string("Para entender la historia de FNAF")) << std::endl;

    std::vector<int> lista = {1, 2, 3};
    std::cout << "Valores en la lista: ";
    for (const auto& elem : lista) {
        std::cout << retornarValor(elem) << " ";
    }
    std::cout << std::endl;

    return 0;
}
